core/sys_context.go

```diff
@@ -7,27 +7,46 @@ import (
 	"github.com/celo-org/celo-blockchain/contracts/blockchain_parameters"
 	"github.com/celo-org/celo-blockchain/contracts/currency"
 	"github.com/celo-org/celo-blockchain/contracts/gasprice_minimum"
+	"github.com/celo-org/celo-blockchain/core/state"
+	"github.com/celo-org/celo-blockchain/core/types"
 	"github.com/celo-org/celo-blockchain/core/vm"
 )
 
-// SysContractCallCtx represents a system contract call context for a given block (represented by vm.EVMRunner).
-// It MUST sit on the header.Root of a block, which is parent of the block we intend to deal with.
+// SysContractCallCtx acts as a cache holding information obtained through
+// system contract calls to be used during block processing.
```

Comment:
Maybe also worth mentioning the system contract calls are done based on starting of the block(or end of parent block)?

Reply:
Hey @YoduYodu, I think I covered that with the constructor comment. See here - https://github.com/celo-org/celo-blockchain/pull/1900/files#diff-d591f9889dcb3c67b0f48fd0671a1a40ebaf6edf021c05b50f2a5698ae79fe24R34-R36

If not, can you elaborate a bit?